### PR TITLE
Fixed public properties to have same JSON attributes

### LINKED ISSUE #1111: Fix LastModified type
ThroughputProperties has `public DateTime LastModified`

All the other types have it has nullable
`public DateTime? LastModified`

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

This fixes all the public properties to have the same JSON attributes for the same public property like etag. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues

closes #1111

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs [hunk 1]
```diff
@@ -35,23 +35,23 @@ public class ThroughputProperties
         /// <remarks>
         /// ETags are used for concurrency checking when updating resources.
         /// </remarks>
-        [JsonProperty(PropertyName = Constants.Properties.ETag)]
+        [JsonProperty(PropertyName = Constants.Properties.ETag, NullValueHandling = NullValueHandling.Ignore)]
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs]: Is this breaking today?  or what's the reason to add the NullValueHandling?

File: Microsoft.Azure.Cosmos/src/Resource/Settings/PermissionProperties.cs [hunk 2]
```diff
@@ -173,7 +173,7 @@ internal PermissionProperties()
         /// <remarks>ResourceToken generation and reading does not apply.</remarks>
         [JsonConverter(typeof(UnixDateTimeConverter))]
         [JsonProperty(PropertyName = Constants.Properties.LastModified, NullValueHandling = NullValueHandling.Ignore)]
-        public DateTime? LastModified { get; private set; }
```
[the remark this answers is not in the document]
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/PermissionProperties.cs]: Yes, but it's a breaking change and we haven't had any user complain about. It should be fixed in v4.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/ThroughputProperties.cs]: Half of the properties have the ignore the other don't. It seems like making it consistent between all the types was missed. Searlizing a null _etag or other system properties and sending it to the backend only increase payload size and has no benefit.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/PermissionProperties.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/PermissionProperties.cs]: Please create a tracking issue for V4.

[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/PermissionProperties.cs]: these are system types are using serialiation outside is mostly not a scenario. 
Am signing off.

**Pull request merged**